Cython/Utility/Coroutine.c

```diff
@@ -1501,13 +1501,31 @@ static PyObject *__Pyx_CoroutineAwait_no_new(PyTypeObject *type, PyObject *args,
 }
 #endif
 
+#if PY_VERSION_HEX < 0x03060000 || CYTHON_USE_TYPE_SPECS
```

Comment:
I'm not sure whether this should be in the version-check guards - there's no possibly no harm in just being explicit and forcing it anyway.

Reply:
Not sure either. You could argue both ways – that it's safer to implement it and let it fail, or that having an implementation at all would rather indicate that there is support although there isn't.

Given that we seem to need it together with type specs, and thus, will probably "always" have it available also in future Python versions, I guess removing the entire guard isn't wrong. You can comment it out for now (and add a comment why) to indicate that there is such a relation to Python versions.

Reply:
I've done that. It looks like `object` provides a default implementation so I don't think that the presence of an implementation indicates anything really